lib/Search/SearchQuery.php

```diff
@@ -154,39 +199,67 @@ private static function filterArray(array $array, $key, $default, $filter = null
      *
      * @return int
      */
-    public function getFrom()
+    public function getFrom(): int
     {
-        return (int)$this->from;
+        return $this->from;
     }
 
     /**
      * The size of the search results.
      *
      * @return int
      */
-    public function getSize()
+    public function getSize(): int
     {
-        if ((int)$this->size < 0) {
+        if ($this->size < 0) {
             $this->size = 1;
         }
-        return (int)$this->size;
+
+        return $this->size;
+    }
+
+    /**
+     * Get the default engine by checking the config
+     *
+     * @return null|string
+     */
+    public function getDefaultEngine(): ?string
+    {
+        global $sugar_config;
+
+        if (!empty($sugar_config['search']['defaultEngine'])) {
```

Comment:
Hi @Dillon-Brown, what should this method return if `$sugar_config['search']['defaultEngine']` is empty

Reply:
Changed to return ```BasicSearchEngine``` when the config is not set. I think that would make the most sense since it's the only engine that doesn't require any setup and works OOTB.